################################################################################
SHORT DESCRIPTION:
################################################################################
Allow tapdisk to be a cdrom backend

################################################################################
LONG DESCRIPTION:
################################################################################

################################################################################
CHANGELOG
################################################################################
Derived from original patch by:
Jed Lejosne <lejosnej@ainfosec.com>

################################################################################
REMOVAL
################################################################################
Whenever things get fixed upstream

################################################################################
UPSTREAM PLAN
################################################################################
We should try to upstream this

################################################################################
INTERNAL DEPENDENCIES
################################################################################

################################################################################
PATCHES
################################################################################
--- a/tools/libxl/libxl_device.c
+++ b/tools/libxl/libxl_device.c
@@ -326,11 +326,8 @@ static int disk_try_backend(disk_try_bac
         if (libxl_defbool_val(a->disk->colo_enable))
             goto bad_colo;
 
-        if (a->disk->is_cdrom) {
-            LOG(DEBUG, "Disk vdev=%s, backend tap unsuitable for cdroms",
-                       a->disk->vdev);
-            return 0;
-        }
+        /* Note: keep going if the disk is a cdrom. tapdisk can handle them as "raw" */
+
         if (!libxl__blktap_enabled(a->gc)) {
             LOG(DEBUG, "Disk vdev=%s, backend tap unsuitable because blktap "
                        "not available", a->disk->vdev);
